stages/org.osbuild.chrony

```diff
@@ -1,11 +1,38 @@
 #!/usr/bin/python3
 """
-Configure chrony to set system time from the network
-
-Configures `chrony` to set the system time from the given `timeservers`.
-
-Modifies /etc/chrony.conf, removing all "server" or "pool" lines and adding
-a "server" line for each server listed in `timeservers`.
+Configure chrony to set system time from the network.
+
+Configures `chrony` by modifying `/etc/chrony.conf`.
+
+Before new values are added to the chrony configuration, all lines starting with
+"server", "pool" or "peer" are removed.
+
+The 'timeservers' option provides a very high-level way of configuring chronyd
+with specific timeservers. Its value is a list of strings representing the
+hostname or IP address of the timeserver. For each list item, the following
+line will be added to the configuration:
+`server <HOSTNAME/IP> iburst`
+
+The 'servers' options provides a direct mapping to the `server` directive from
+chrony configuration. Its value is a list of dictionaries representing each
+timeserver which should be added to the configuration. For each list item,
+a `server` directive will be added the configuration. Currently supported
+subset of options which can be specified for each timeserver item:
+  - 'hostname' (REQUIRED)
+  - 'minpoll'
+  - 'maxpoll'
+  - 'iburst' (defaults to true)
+  - 'prefer' (defaults to false)
+
+The 'leapsectz' option configures chrony behavior related to automatic checking
+of the next occurrence of the leap second, using the provided timezone. Its
+value is a string representing a timezone from the system tz database (e.g.
+'right/UTC'). If an empty string is provided, then all occurrence of 'leapsectz'
+directive are removed from the configuration.
+
+Constrains:
+  - At least one of the 'timeservers' or 'servers' option must be provided.
+  - 'timeservers' option can not be combined with 'servers' option.
```

Comment:
I think this can be simplified to one constraint: Exactly one of `timeservers` or `servers` options must be provided.
But keeping it as two points is also fine.

Reply:
Yeah, my expectation is that in the future we may add also 'pool' and 'peer' directives, but I can definitely simplify the sentence for now...